src/bin/server.rs

```diff
@@ -94,13 +107,26 @@ async fn search(
         searcher: &Searcher,
         dir: &Dir,
     ) -> Result<Html<String>, ServerError> {
-        let (count, docs) = searcher.search(&params.query)?;
+        if params.page == 0 || params.results_per_page == 0 {
```

Comment:
There should also be a maximum accept value for results_per_page to avoid denial of service.